### DISCUSSION TITLE
Docker CPU and Memory metrics via prometheus_exporter

### DISCUSSION THREAD

**Discussion by lzwaan** [vectordotdev/vector#13747] [Q&A] [answered]
Hi,

I'm looking for a way to monitor cpu and memory usage by vector which run in a docker container. The metrics for in -and output is already very useful, but I can't find a solution yet for simple cpu and memory, I tried the host metrics but then it will show only a total of my server.
I was hoping to select via a dropdown in Grafana, so if anyone has suggestions, hints that would be great!
Next week we like to test the performance of vector encrypting SSIDs from modem data till we reach around 500.000 messages per minute, so we like to monitor the pipeline if it keeps up in speed without discarding / dropping messages...

**Comment by jszwedko** [comment, marked as answer]
Hi @lzwaan !

You are correct that Vector doesn't currently publish metrics about its process's CPU and memory utilization. Would you want to open a feature request? As a workaround you can externally monitor the Vector process with something like https://github.com/ncabatoff/process-exporter and have Vector scrape that with the `prometheus_scrape` source.
